src/libexpr/eval.cc

```diff
@@ -973,22 +973,20 @@ inline Value * EvalState::lookupVar(Env * env, const ExprVar & var, bool noEval)
 
     if (!var.fromWith) return env->values[var.displ];
 
+    if (noEval) return nullptr;
+
+    auto * fromWith = var.fromWith;
     while (1) {
-        if (env->type == Env::HasWithExpr) {
-            if (noEval) return 0;
-            Value * v = allocValue();
-            evalAttrs(*env->up, (Expr *) env->values[0], *v, noPos, "<borked>");
-            env->values[0] = v;
-            env->type = Env::HasWithAttrs;
-        }
+        forceAttrs(*env->values[0], noPos, "<borked>");
```

Comment:
Wasn't your message, but in case it does pop up:

```suggestion
        forceAttrs(*env->values[0], noPos, "while evaluating the first subexpression of a with expression");
```

Reply:
Sounds good